File: weather-demo-app-apple/Weather/UI/Current/CurrentWeatherSuccessView.swift
//
//  CurrentWeatherSuccessView.swift
//  Weather
//
//  Created by Onur on 4/11/26.
//

import SwiftUI

struct CurrentWeatherSuccessView: View {
    let weather: CurrentWeather
    let onForecastClicked: () -> Void

    var body: some View {
        VStack(alignment: .leading) {
            HStack(spacing: 12) {
                IconView(iconCode: weather.iconCode, size: 72)

                VStack(alignment: .leading) {
                    Text(weather.city)
                        .font(.title3)

                    Text(weather.description.capitalized)
                        .font(.body)
                }
            }
            
            Spacer().frame(height: 16)

            Text("\(weather.temperatureCelsius, specifier: "%.1f")°C")
            Text("Feels like \(weather.feelsLikeCelsius, specifier: "%.1f")°C")
            
            Spacer().frame(height: 16)

            Text("Humidity: \(weather.humidityPercent)%")
            Text("Wind speed: \(weather.windSpeedKph, specifier: "%.1f") km/h")
            
            Spacer().frame(height: 20)

            Button("See forecast", action: onForecastClicked)
                .buttonStyle(.borderedProminent)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(16)
        .background(.thinMaterial)
        .clipShape(RoundedRectangle(cornerRadius: 16))
    }
}

File: weather-demo-app-android/app/src/main/java/com/demo/weather/ui/current/CurrentWeatherSuccessView.kt
package com.demo.weather.ui.current

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.capitalize
import androidx.compose.ui.unit.dp
import com.demo.weather.domain.model.CurrentWeather
import com.demo.weather.ui.common.IconView
import java.util.Locale

@Composable
fun CurrentWeatherSuccessView(
    weather: CurrentWeather,
    onForecastClicked: () -> Unit
) {
    Card(
        modifier = Modifier.fillMaxWidth()
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp)
        ) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                IconView(
                    iconCode = weather.iconCode,
                    contentDescription = weather.description,
                    size = 72.dp
                )

                Column {
                    Text(
                        text = weather.city,
                        style = MaterialTheme.typography.headlineSmall
                    )

                    Text(
                        text = weather.description.capitalize(Locale.getDefault()),
                        style = MaterialTheme.typography.bodyLarge
                    )
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            Text("${"%.1f".format(Locale.US, weather.temperatureCelsius)}°C")
            Text("Feels like ${"%.1f".format(Locale.US, weather.feelsLikeCelsius)}°C")

            Spacer(modifier = Modifier.height(16.dp))

            Text("Humidity: ${weather.humidityPercent}%")
            Text("Wind speed: ${"%.1f".format(Locale.US, weather.windSpeedKph)} km/h")

            Spacer(modifier = Modifier.height(20.dp))

            Button(
                onClick = onForecastClicked
            ) {
                Text("See forecast")
            }
        }
    }
}